controller/readwrite.go

```diff
@@ -328,6 +328,8 @@ func (rw *ReadWrite) checkApply(ctx context.Context, d driver.Driver, retry, onc
 	// new data
 	if rendered {
 		rw.logger.Info("executing task", taskNameLogKey, taskName)
+		rw.drivers.SetActive(taskName)
+		defer rw.drivers.SetInactive(taskName)
```

Comment:
Just verifying my understanding.

We set it to active while running, so that now, if we call `delete` via the api, we cannot delete an `active` task correct?

Reply:
Yeah, there are a lot of complications around deleting a task when it's running, such as:

- The request will hang if until the task is complete
- At the end of the task running, an event is stored, and all events for a task need to be deleted when the task is deleted
- When we add cleanup in the future, we may clean something up but the running task might just add it back

So I made it where it returns the 409 Conflict when we delete an active, running task.